### PR TITLE
dart: fix required params.  add 'multi' query params.  better deserialization.

### PR DESCRIPTION
**Pull request by inc-cloitsch**
see commit messages for more details

### CODE REVIEW

**Comment by wing328** [comment]
Thanks for the PR but your commit (as shown in the Commits tab) is not linked to your Github account, which means this PR won't count as your contribution in https://github.com/swagger-api/swagger-codegen/graphs/contributors.

Let me know if you need help fixing it.

Ref: http://stackoverflow.com/questions/3042437/change-commit-author-at-one-specific-commit

File: bin/dart-petstore.sh [hunk 1]
```diff
@@ -28,4 +28,7 @@ fi
 export JAVA_OPTS="${JAVA_OPTS} -XX:MaxPermSize=256M -Xmx1024M -DloggerPath=conf/log4j.properties"
 ags="$@ generate -i modules/swagger-codegen/src/test/resources/2_0/petstore.yaml -l dart -o samples/client/petstore/dart"
 
+# for dart vm lib generation:
+#ags="$@ generate -i modules/swagger-codegen/src/test/resources/2_0/petstore.yaml -l dart -o samples/client/petstore/dart --additional-properties browserClient=false"
+
```
Comment (wing328) [inline bin/dart-petstore.sh]: You may want to create another dart petstore shell script (e.g. dart-vmlib-petstore.sh) for the `--additional-properties browserClient=false` option (similar to what we've done for other langauges, e.g. java)

Comment (wing328) [inline bin/dart-petstore.sh]: And you may consider later adding a CLI option (e.g. dartBrowserClient) instead of using additional properties.

**Comment by close2** [comment]
Hi,

(from another account)  I don't really care about being listed in the contributers file.

I have opened an issue for the cli option and dartvm shell script.

**Comment by wing328** [comment]
@close2 the commit authorship is fixed. Thanks for the PR, which has been merged into master.

**Pull request merged**